src/pybind/mgr/cephadm/module.py

```diff
@@ -2535,10 +2535,42 @@ def _apply(self, spec: GenericSpec) -> str:
 
         return self._apply_service_spec(cast(ServiceSpec, spec))
 
+    def _get_candidate_hosts(self, placement: PlacementSpec) -> List[str]:
+        """Return a list of candidate hosts according to the placement specification."""
+        all_hosts = self.cache.get_schedulable_hosts()
+        draining_hosts = [dh.hostname for dh in self.cache.get_draining_hosts()]
+        ls = []
+        if placement.hosts:
+            ls = [h.hostname for h in placement.hosts
+                  if h.hostname in placement.hosts and h.hostname not in draining_hosts]
```

Comment:
in this case, we actually want to not match to draining hosts regardless of whether the placement used explicit hosts, labels pattern etc. In the scheduling for actual services this was handled elsewhere but in this case we will have to handle it explicitly. We can probably remove this `not in draining_hosts` here and instead just apply that at the end before returning so it isn't dependent on what method the spec is using (explicit, labels, pattern etc.)

Reply:
fixed